src/StreamMaster.hpp

```diff
@@ -21,31 +26,39 @@ struct FileWriterCommand;
 
 namespace FileWriter {
 
-template <typename Streamer, typename Demux> struct StreamMaster {
+template <typename Streamer, typename Demux> class StreamMaster {
+  using Error = Status::StreamMasterErrorCode;
+  using Options = typename Streamer::Options;
 
-  StreamMaster() : do_write(false), _stop(false){};
+public:
+  StreamMaster(){};
 
-  StreamMaster(
-      std::string &broker, std::vector<Demux> &_demux,
-      std::vector<std::pair<std::string, std::string>> kafka_options = {},
-      const RdKafkaOffset &offset = RdKafkaOffsetEnd)
-      : demux(_demux), do_write(false), _stop(false) {
+  StreamMaster(std::string &broker, std::vector<Demux> &_demux,
+               const Options &kafka_options = {},
+               const Options &filewriter_options = {})
+      : demux(_demux) {
 
     for (auto &d : demux) {
-      streamer.emplace(d.topic(), Streamer{broker, d.topic(), kafka_options});
+      streamer.emplace(std::piecewise_construct,
+                       std::forward_as_tuple(d.topic()),
+                       std::forward_as_tuple(broker, d.topic(), kafka_options,
+                                             filewriter_options));
       streamer[d.topic()].n_sources() = d.sources().size();
     }
   };
 
-  StreamMaster(
-      std::string &broker, std::unique_ptr<FileWriterTask> file_writer_task,
-      std::vector<std::pair<std::string, std::string>> kafka_options = {},
-      const RdKafkaOffset &offset = RdKafkaOffsetEnd)
-      : demux(file_writer_task->demuxers()), do_write(false), _stop(false),
+  StreamMaster(std::string &broker,
```

Comment:
Minor thing, but if taken by `&`, it could be `const` here as well.

Reply:
:white_check_mark: